### DISCUSSION TITLE
Creating Several Properties at Once

### DISCUSSION THREAD

**Discussion by arenner87** [Gremlinq/ExRam.Gremlinq#705] [Q&A]
Hi, 

Introduced in TinkerPop 3.6 (April 2022) was the ability to add multiple Properties as a map of key/value pairs - notes linked below. Is this possible via Gremlinq yet (or planned for a future release)?

Thanks in advance!

https://github.com/apache/tinkerpop/blob/master/CHANGELOG.asciidoc

"Extended property()` to allow for setting a Map of property value"

https://kelvinlawrence.net/book/Gremlin-Graph-Guide.html

"The property step can now be given a map of key/value pairs so that several properties can be created at once."

**Comment by danielcweber** [comment]
What would a possible improvement for Gremlinq-syntax look like?

**Reply by arenner87** [reply]
Assuming you mean this is already possible with the current syntax, I have successfully added a dynamic list of Properties by delaying awaiting the IVertexGremlinQuery, allowing me to loop through the input list, adding each Key/Value as an individual Property, thus keeping RUs to a minimum.

While this might not be the most optimal solution (potentially), im happy with it for now. Thanks.
